catalog/data_volume.go

```diff
@@ -0,0 +1,27 @@
+package catalog
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/catalog"
+	"github.com/databricks/terraform-provider-databricks/common"
+)
+
+func DataSourceVolume() common.Resource {
+	return common.WorkspaceData(func(ctx context.Context, data *struct {
+		Id     string              `json:"id,omitempty" tf:"computed"`
+		Name   string              `json:"name"`
+		Volume *catalog.VolumeInfo `json:"volume_info,omitempty" tf:"computed"`
+	}, w *databricks.WorkspaceClient) error {
+		fmt.Println(data)
+		volume, err := w.Volumes.ReadByName(ctx, data.Name)
+		if err != nil {
+			return err
+		}
+		data.Volume = volume
+		data.Id = volume.VolumeId
```

Comment:
For unification with the Volume resource, it's better to set it to the `name`: https://registry.terraform.io/providers/databricks/databricks/latest/docs/resources/volume#id—the `VolumeId` is an internal UUID.

Reply:
set to fully qualified name (`FullName`) in 714d0c2432d14fdfcf765f9d5204db09936f29c1